### PR TITLE
adding hardware documentation #7

### PR DESCRIPTION
**Pull request by GG23800**

### CODE REVIEW

**Comment by Aurelie-Mutschler** [comment]
ping @Bivi could you please review the hardware documentation @jerome2echopen just made? we need some feedbacks, for example are there enough details or should we add some information ?

**Comment by Bivi** [comment]
For me the second diagram is confusing : in french, very similar to the first one and false (we never decided to control directly the TGC with a smartphone ...). It would be simpler and more usefull to choose only one diagram (even if the first one is functional and the second a design diag) and add one paragraph for each component containing all data concerning this component and links with others, . Example :

     ...
## TGC
Time Gain Compensation, is an electronic component used to compensate signal attenuation during time.  TGC output is theoretically in range +/- 2.5 V, generally it is between +/- 1V....bla bla bla. TGC amplifies signal from ... TGC is controlled by ... TGC output is used by ...
### TGC caracteristic:
Amplification command: between 0 and 1 V
Amplification on LO gain mode: from -4.5 to +43.5 dB
Amplification on HI gain mode: from +7.5 to +55.5 dB
Output: between -2.5 and +2.5 V
### TGC Settings
TGC: LO gain mode (less amplification but too much noise on HI gain mode)
TGC amplication command: between 0.3 and 1 V (so amplification is between 10 and 43.5 dB)
   ...

**Comment by Aurelie-Mutschler** [comment]
thanks @Bivi !
I wait for @jerome2echopen to update the file before I merge the PR. I'll have a look at it more in details by the beginning of next week.

**Comment by GG23800** [comment]
Yep, the synoptic diagram was wrong, I forgot to made all modifications needed.

I changed the order of the diagram with few description so one can look at one or the other. If you still think it is redundant, we will clear it.

I don't agree with the vision of describing all modules one by one like this. It is not the technical documentation of the hardware, it is a more general discussion about the device. And I think it preferable to put all the settings together so one don't have to navigate in the document to find all of them.

I make a new pull request.

File: stable/characteristics.md [hunk 1]
```diff
@@ -1 +1,33 @@
...
+* 60° sector image
+* 1 mm radial resolution
+* lateral resolution is unknowed yet
+* 1 frequency (3.5 MHz)
+
+## Informations
+
+In our device number of lines, sector angle, beggining of measurement, end of measurement and amplification factor are variables.
+
+By default we use a 60° sector, 64 lines and measure from 80 to 160 mm depth. We these settings, we have 1689 points per lines. So an image is composed of 108096 points (the sector image can be included in a rectangle 91 mm *160 mm).
```
Comment (Aurelie-Mutschler) [inline stable/characteristics.md]: I don't understand how we deduce the number of points per line from the previous sentence ?! The number of points is just related to the listening time and sampling rate, right? Why is the number of points constant for all depth values ? Does that mean that you change the sampling rate ?!

File: stable/characteristics.md [hunk 2]
```diff
@@ -1 +1,33 @@
...
+
+The device is used as an access point and a terminal must connect to it. The device send the RAW data encoded on 16 bits *via* TCP protocol to the terminal (so an image has a size of 4.2 Mbits). The device send the data line by line (first 16 bits of the buffer give the number of the line). The device can send the data to multiple terminals at the same time.
```
Comment (Aurelie-Mutschler) [inline stable/characteristics.md]: The information about image size  is relevant here, but the description of the data transfer between device and terminal should be in doc_hardware.md

File: stable/characteristics.md [hunk 3]
```diff
@@ -1 +1,33 @@
...
+
+For making an image, the terminale receive the data, for each line it determines the envelope of the signal with an Hilbert transform. When all the lines of an image are received and processed, the terminal displays the image on gray scale as a sector image.
```
Comment (Aurelie-Mutschler) [inline stable/characteristics.md]: Similarly, this paragraph is a description of the functionning, not a characteristic, it shouldn't be here.
+ it's incomplete, the scan conversion isn't mentioned. Anyway, this should be in the software documentation, so just delete this paragraph please

File: stable/characteristics.md [hunk 4]
```diff
@@ -1 +1,33 @@
...
+
+* around 3 frames per second
+* 128 gray level
+* 64-128 lines
+* 60° sector image
+* 1 mm radial resolution
+* lateral resolution is unknowed yet
+* 1 frequency (3.5 MHz)
+
+## Informations
```
Comment (Aurelie-Mutschler) [inline stable/characteristics.md]: This section should include the following pieces of information :
- recall the default tunings of the device --> There should be a **subsection "Default tunings"** where you recall these informations
- depth of detection, angular coverage --> the paragraph about depth of detection is not clear to me ? + these informations should be in a **subsection "Detection characteristics"**
- SNR --> can you quantify this + add the nice figures you made to visualize the last improvement? --> in a **subsection "Raw signal characteristics"**
- number of pixels, size of pixels, framerate --> can you gather those informations together with the image size, in a unique **subsection "Display characteristics"** ? 
- add a screenshot of a beautiful echOpen image :-)

File: stable/doc_hardware.md [hunk 5]
```diff
@@ -1 +1,107 @@
-coming soon...
\ No newline at end of file
+# Hardware documentation
+
+In this section we present a short documentation of our kit. More informations about the hardware can be found in our [website](http://echopen.org/doc-website/CAT-configuration/CFG-sweeping_probe/content.html) or download our [github folder](https://github.com/echopen/PRJ-medtec_kit), aquarium configuration (PRJ-medtec_kit/configuration/CFG-aquarium_kit/readme.md). We show synoptic and block diagrams of the kit and give some informations about daughter boards and modules.
```
Comment (Aurelie-Mutschler) [inline stable/doc_hardware.md:3]: - The link to the documentation website points to the old configuration --> @Bivi when will the website be updated with the new configuration ?

- On the repo, the config "Aquarium" is v2.1.0, whereas it's supposed to be v3.0.0

File: stable/doc_hardware.md [hunk 6]
```diff
@@ -1 +1,107 @@
...
+
+### Synoptic diagram
```
Comment (Aurelie-Mutschler) [inline stable/doc_hardware.md]: The synoptic diagram seems incomplete. For example, where is the high voltage ? And the WIFI ?

File: stable/doc_hardware.md [hunk 7]
```diff
@@ -1 +1,107 @@
...
+
+Here is a simple diagram presenting the general functions of the device.
+
+![](./doc_hardware/synoptic_diagram2.png)
+
+
+### Block diagram
```
Comment (Aurelie-Mutschler) [inline stable/doc_hardware.md]: The bloc diagram is not up-to-date as it still shows the envelope detection, but it's not implemented on the device. 
Besides, I agree with @Bivi remark about the two diagrams. Let's get rid of this bloc diagram here, @clecoued and I will do a more general one that describes the device + app pipeline.
Then, the synoptic diagram should be more exhaustive (see first sentence in this comment)

File: stable/doc_hardware.md [hunk 8]
```diff
@@ -1 +1,107 @@
...
+
+![](./doc_hardware/block_diagram.png)
+
+The device generate a high voltage pulse to excite the transducer (*via* or not an impedance matching layer). The rest of the device is protected from the high voltage pulse by a T/R switch. The signal is amplified by a TGC (variable amplification driven by the controller), filtered (pass-band filter) and converted to digital signal. The signal is the send *via* wifi to a terminal for display.
+
+For making a 2D image, the transducer must make a sweeping movement. The transducer is placed on the choosen line by the controller, acquisition is made and send to the terminal, then the transducer is moved to the next line. The lines are angulary equally reparted on the sector of measurement.
+
+## Device informations
+
+### Acoustic
```
Comment (Aurelie-Mutschler) [inline stable/doc_hardware.md]: Can you move the "### Acoustic" section to characteristics.md, in a dedicated "Transducer characteristics" section ?

File: stable/doc_hardware.md [hunk 9]
```diff
@@ -1 +1,107 @@
...
+
+We use a commercial transducer from imasonic with a central frequency of 3.27 MHz. Technical informations about the transducer (temporal and frequential pulse respectively) can be found in the following images
+
+![](./doc_hardware/signalt.png "Temporal pulse of the transducer")
```
Comment (Aurelie-Mutschler) [inline stable/doc_hardware.md]: Can you at least put a legend to describe the content of the figures ?

File: stable/doc_hardware.md [hunk 10]
```diff
@@ -1 +1,107 @@
...
+
+Transducer characteristics:
+
+* Pulse duration at -6 dB: 382 ns (precision of 0.56 mm)
+* Acoustic pass-band: 2.13 MHz, from 2.21 to 4.33 MHz
+* Focal depth: 120 mm
+* Excitation time: around half the period = 142 ns
+* Excitation voltage: < -150 V
+
+### Electronic (analogic)
```
Comment (Aurelie-Mutschler) [inline stable/doc_hardware.md]: Here you mix description and characteristics. Is it possible to keep only the description in doc_hardware.md and move the signal characteristics in characteristics.md?
Anyway, this section is not clear, it should be at least more structured

File: stable/doc_hardware.md [hunk 11]
```diff
@@ -1 +1,107 @@
...
+### Acquisition
+
+Acquisition is made with a RedPitaya which caracteristics are:
+
+* Sampling rate: 125 or 125/8 Msps
+* Precision: 14 bits
+* Voltage range: +/- 1 V or +/- 20 V
+* Buffer maximum size: 16384 points
+
+Considering that the acoustic waves makes back and forth between the transducer and the medium, depth of measurement is given by d=v*t/2. So maximum depth of measurement at 125 Msps is **97 mm** and at 125/8 Msps it is **775 mm**.
```
Comment (Aurelie-Mutschler) [inline stable/doc_hardware.md]: The discussion about maximal depth should rather be in characteristics.md.
Besides, I don't understand at all the explanations given here. What's the link between sampling rate and depth of detection ?! Why do the value of the sampling rate influences the maximal depth ?!

**Review by Aurelie-Mutschler — CHANGES_REQUESTED** [review]
Here are some comments about the hardware documentation and the device characteristics.

File: stable/doc_hardware.md (hunk 6 above)
Comment (Aurelie-Mutschler) [reply stable/doc_hardware.md]: discussed with @jerome2echopen : let's make a diagram focused on modules. The functional diagrams (general diagrams + pipeline of data processing) will be displayed in the high-level documentation

File: stable/doc_hardware.md [hunk 12]
```diff
@@ -1 +1,123 @@
...
+* Pulser: [MDL-pulser_v2](https://github.com/echopen/PRJ-medtec_kit/blob/master/electronic/modules/hardware/MDL-pulser/MDL-pulser_v2/readme.md)
+* Transducer: [MDL-transducer_v4](https://github.com/echopen/PRJ-medtec_kit/tree/master/electronic/modules/hardware/MDL-transducer/MDL-transducer_v4)
+* Supply: [MDL-power_supply_v1](https://github.com/echopen/PRJ-medtec_kit/blob/master/electronic/modules/hardware/MDL-supply/MDL-supply_v1/readme.md)
+* T/R switch: [MDL-tr_switch](https://github.com/echopen/PRJ-medtec_kit/blob/master/electronic/modules/hardware/MDL-tr_switch/MDL-tr_switch_v1/readme.md)
+* TGC: [MDL-tgc_v1](https://github.com/echopen/PRJ-medtec_kit/blob/master/electronic/modules/hardware/MDL-tgc/MDL-tgc_v1/readme.md)
+* Filter: [MDL-filter_v1](https://github.com/echopen/PRJ-medtec_kit/blob/master/electronic/modules/hardware/MDL-filter/MDL-filter_v1/readme.md)
+* Envelope detection: [MDL-envelope_detector_v1](https://github.com/echopen/PRJ-medtec_kit/blob/master/electronic/modules/hardware/MDL-envelope_detector/MDL-envelope_detector_v1/readme.md)
```
Comment (Aurelie-Mutschler) [inline stable/doc_hardware.md]: as discussed, let's remove the envelope detection module ;-)

**Comment by Aurelie-Mutschler** [comment]
@jerome2echopen thanksss the doc_hardware.md is awesome !! ;-)
there's still a link to the envelope detection module that has to be dismissed. Otherwise I think we're good with doc_hardware.md
Waiting for updates of characteristics.md to merge the PR... to be continued !

**Pull request merged**